### DISCUSSION TITLE
How can I get the DDS time with OpenDDS in Java

### DISCUSSION THREAD

**Discussion by yuan-ge** [OpenDDS/OpenDDS#4629] [Q&A] [answered]
I'm using OpenDDS with Java17, and now, I want to get the DDS time to compare with the timestamp in SampleInfo, so that I can know the transmission delay from pub to sub.
I have read the doc of opendds, and I just find this:
![image](https://github.com/OpenDDS/OpenDDS/assets/61228924/581938e4-7217-46f9-a6f8-c2d3b6f0338d)
in OpenDDS Development Guidelines. But I didn't find the corresponding class in DDS library.
Is there any way to solve this problem? Thanks.

**Comment by iguessthislldo** [comment, marked as answer]
`SystemTimePoint` is an internal C++ type. You can get the current `Time_t` using the `get_current_time` method on your `DomainParticipant`. It's just Unix epoch time, so you can also get the same exact information from the Java standard library using something like `getEpochSecond()` and `getNano()` from `Instant.now()`.

**Reply by yuan-ge** [reply]
Is this `Time_t` get by `get_current_time`  the synchronization time of DDS system?

**Reply by jrw972** [reply]
`Time_t` is the type used by DDS to represent points in time such as when samples were sent and other time points.  It is usually derived from the system clock so it assumes that the clocks of both writer and reader are synchronized.
